Client/Frontend/Browser/TabManager.swift

```diff
@@ -166,6 +160,29 @@ class TabManager: NSObject {
         assert(Thread.isMainThread)
         return allTabs.filter { $0.type == type }
     }
+    
+    private class func getNewConfiguration(isPrivate: Bool) -> WKWebViewConfiguration {
+        let configuration = WKWebViewConfiguration()
+        configuration.processPool = WKProcessPool()
+        configuration.preferences.javaScriptCanOpenWindowsAutomatically = !Preferences.General.blockPopups.value
+        return configuration
+    }
+    
+    private func resetConfiguration() {
+        configuration = TabManager.getNewConfiguration(isPrivate: false)
+        privateConfiguration = TabManager.getNewConfiguration(isPrivate: true)
+    }
+    
+    func reset() {
+        resetConfiguration()
+        allTabs.filter({$0.webView != nil}).forEach({
+            $0.resetWebView(config: $0.isPrivate ? privateConfiguration : configuration)
+        })
+        selectTab(selectedTab, ignorePrevious: true)
```

Comment:
Well actually no, If the user enables block all cookies then we have to reload the selectedTab. Other tabs load on tab selection but the currently selectedTab needs reload call explicitly. Unless this bool is there while previous and selectedTab are nil, the underlying exit condition will be met and the func will return(without reloading the tab). This logic is in place to avoid the func execution in case the current tab is selected. This function is tightly coupled with the tabmanger logic. I don't find it helpful to change so much in this PR. Maybe we can make more loosely coupled funcs in refactor.

Reply:
What I'm saying is that this `ignorePrevious` is only used in this one case to select the already selected tab.

You could just do this:

```swift
let currentSelectedTab = selectedTab // Retain ref to the tab you want to "re"select
_selectedTab = -1 // Reset selected tab (now `selectedTab` returns nil)
selectTab(currentSelectedTab) // Select it
```

Reply:
Oh yeah, this solves it! Thanks :)